frontend/src/API.js

```diff
@@ -1,31 +1,49 @@
 import axios from "axios";
 
 var baseURL;
-if (process.env.REACT_APP_ENVIRONMENT && process.env.REACT_APP_ENVIRONMENT === "PRODUCTION") {
+if (
+    process.env.REACT_APP_ENVIRONMENT &&
+    process.env.REACT_APP_ENVIRONMENT === "PRODUCTION"
+) {
     baseURL = process.env.REACT_APP_API_BASE_URL;
 } else {
     baseURL = "http://127.0.0.1:8000";
 }
 
 const api = axios.create({
-  baseURL: baseURL,
-  headers: {
-    "Content-Type": "application/json",
-  },
+    baseURL: baseURL,
+    headers: {
+        "Content-Type": "application/json"
+    },
 });
 
-export default class API {
+api.interceptors.request.use(
+    config => {
+        if (config.requireToken) {
+            const user = localStorage.getItem('LOGIN_USER_KEY') ? JSON.parse(localStorage.getItem('LOGIN_USER_KEY')) : null;
```

Comment:
Define LOGIN_USER_KEY on the page top.

const LOGIN_USER_KEY = "E_LIKE_ LOGIN_USER_KEY";